webapp/components/Chat/Chat.vue

```diff
@@ -147,20 +162,34 @@ export default {
             query: messageQuery(),
             variables: {
               roomId: room.id,
+              first: this.messagePageSize,
+              offset,
             },
             fetchPolicy: 'no-cache',
           })
-          this.messages = Message
+
+          const msgs = []
+          ;[...this.messages, ...Message].forEach((m) => {
+            msgs[m.indexId] = m
+          })
+          this.messages = msgs.filter(Boolean)
```

Comment:
What is this?

Reply:
This filters out undefined indexes.
When you request message ids 35,36,37,38,39,40 and put them into an array the js engine seems to add the elements 0-34 as undefined. The library requires integer indexes, to properly work we need stable indexes, the order of the messages is defined by the index, newest messages have a higher ids.

The `map.filter(Boolean)` removes all undefined elements (all not truthy)